orc/src/test/java/org/apache/iceberg/orc/TestExpressionToSearchArgument.java

```diff
@@ -262,4 +264,65 @@ public void testEvolvedSchema() {
     actual = ExpressionToSearchArgument.convert(boundFilter, readSchema);
     Assert.assertEquals(expected.toString(), actual.toString());
   }
+
+  @Test
+  public void testOriginalSchemaNameMapping() {
+    Schema originalSchema = new Schema(
+        required(1, "int", Types.IntegerType.get()),
+        optional(2, "long", Types.LongType.get())
+    );
+
+    TypeDescription orcSchemaWithoutIds = ORCSchemaUtil.removeIds(ORCSchemaUtil.convert(originalSchema));
+    NameMapping nameMapping = MappingUtil.create(originalSchema);
+
+    TypeDescription readSchema = ORCSchemaUtil.buildOrcProjection(originalSchema,
+        ORCSchemaUtil.applyNameMapping(orcSchemaWithoutIds, nameMapping));
+
+    Expression expr = and(equal("int", 1), equal("long", 1));
+    Expression boundFilter = Binder.bind(originalSchema.asStruct(), expr, true);
+    SearchArgument expected = SearchArgumentFactory.newBuilder()
+        .equals("`int`", Type.LONG, 1L)
+        .equals("`long`", Type.LONG, 1L)
+        .build();
+
+    SearchArgument actual = ExpressionToSearchArgument.convert(boundFilter, readSchema);
+    Assert.assertEquals(expected.toString(), actual.toString());
+  }
+
+  @Test
+  public void testModifiedSchemaNameMapping() {
```

Comment:
can we have aa test case for where complex fields are also pruned? For instance if all fields of a struct are pruned, the struct is pruned. Similarly for maps and lists?

Reply:
I think that test is https://github.com/apache/iceberg/pull/1140/files#diff-27a14477d32c8e2921027e1353a76753R216 and does not seem related to test the name mapping as that'd test `buldOrcProjection`. I can definitely add those cases you mention on that test.

Reply:
I've added a couple tests to address this.